### PR TITLE
made temp dir path configurable

### PR DESCRIPTION
**Pull request by jstucke**
The path for Temporary Directories is now configurable in the configuration file (`main.cfg`). This can be useful when the user running FACT doesn't have full access to `/tmp`

### CODE REVIEW

File: src/helperFunctions/config.py [hunk 1]
```diff
@@ -41,3 +42,7 @@ def read_list_from_config(config_file: ConfigParser, section: str, key: str, def
     if not config_entry:
         return default
     return [item.strip() for item in config_entry.split(',') if item]
+
+
+def get_temp_dir_path(config: Optional[ConfigParser] = None) -> str:
+    return config.get('data_storage', 'temp_dir_path', fallback='/tmp') if config else '/tmp'
```
[the remark this answers is not in the document]
Comment (weidenba) [reply src/helperFunctions/config.py:48]: I would recommand to create the tmp dir if it does not exist

File: src/helperFunctions/config.py [hunk 2]
```diff
@@ -13,10 +14,10 @@ def load_config(config_file_name):
     '''
     config = configparser.ConfigParser()
     config_path = '{}/{}'.format(get_config_dir(), config_file_name)
-    if os.path.exists(config_path):
-        config.read(config_path)
-        return config
-    complete_shutdown('config file not found: {}'.format(config_path))
+    if not os.path.exists(config_path):
```
[the remark this answers is not in the document]
Comment (jstucke) [reply src/helperFunctions/config.py:17]: I only inverted the logic to fix the pylint error (inconsistent return) but sure, why not?

File: src/plugins/analysis/file_system_metadata/code/file_system_metadata.py [hunk 3]
```diff
@@ -24,8 +25,8 @@ class MountingError(RuntimeError):
 
 
 @contextmanager
-def mount(file_path, fs_type=''):
-    mount_dir = TemporaryDirectory()
+def mount(file_path, fs_type='', mount_path='/tmp'):
```
[the remark this answers is not in the document]
Comment (jstucke) [reply src/plugins/analysis/file_system_metadata/code/file_system_metadata.py:28]: Not necessarily. Should I remove it?

File: src/plugins/analysis/software_components/internal/resolve_version_format_string.py [hunk 4]
```diff
@@ -14,8 +14,8 @@
 KEY_FILE = 'key_file'
 
 
-def extract_data_from_ghidra(input_file_data: bytes, key_strings: List[str]) -> List[str]:
-    with TemporaryDirectory(prefix='FSR_') as tmp_dir:
+def extract_data_from_ghidra(input_file_data: bytes, key_strings: List[str], path: str = '/tmp') -> List[str]:
```
[the remark this answers is not in the document]
Comment (jstucke) [reply src/plugins/analysis/software_components/internal/resolve_version_format_string.py:17]: Not necessarily. Should I remove it?

File: src/helperFunctions/config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jstucke) [reply src/helperFunctions/config.py:48]: - added an entry to the `main.cfg`
- added `mkdir()` if it doesn't exist
- added test

**Pull request merged**